test/model.test.js

```diff
@@ -0,0 +1,893 @@
+      })
+      .then(function(result) {
+        expect(result).to.have.length(0);
+      });
+  });
+});
+
+describe('virtual', function() {
+  afterEach(function() { return test.dropTables(); });
+
+  it('should pass schema validation', function() {
+    var Model = thinky.createModel(util.s8(), {
+      id: String,
+      num: Number,
+      numVirtual: {
+        _type: 'virtual'
+      }
+    }, { init: false });
+  });
+
+  it('should generate fields', function() {
+    var Model = thinky.createModel(util.s8(), {
+      id: String,
+      num: Number,
+      numVirtual: {
+        _type: 'virtual',
+        default: function() {
+          return this.num + 2;
+        }
+      }
+    }, { init: false });
+
+    var doc = new Model({ num: 1 });
+    expect(doc.numVirtual).to.equal(3);
+  });
+
+  it('should generate fields (manually)', function() {
+    var Model = thinky.createModel(util.s8(), {
+      id: String,
+      num: Number,
+      numVirtual: {
+        _type: 'virtual',
+        default: function() {
+          return this.num + 2;
+        }
+      }
+    }, { init: false });
+
+    var doc = new Model({ num: 1 });
+    expect(doc.numVirtual).to.equal(3);
+
+    doc.num = 2;
+    expect(doc.numVirtual).to.equal(3);
+
+    doc.generateVirtualValues();
+    expect(doc.numVirtual).to.equal(4);
+  });
+
+  it('should validate fields', function() {
+    var Model = thinky.createModel(util.s8(), {
+      id: String,
+      num: Number,
+      numVirtual: {
+        _type: 'virtual'
+      }
+    }, { init: false });
+
+    var doc = new Model({ num: 1 });
+    return doc.validate();
+  });
+
+  it('should not save virtual fields', function() {
+    var Model = thinky.createModel(util.s8(), {
+      id: Number,
+      num: Number,
+      numVirtual: {
+        _type: 'virtual'
+      }
+    });
+
+    var doc = new Model({
+      id: 1,
+      num: 1,
+      numVirtual: 3
+    });
+
+    return Model.ready()
+      .then(function() {
+        return doc.save();
+      })
+      .then(function(result) {
+        return Model.get(1).execute();
+      })
+      .then(function(result) {
+        expect(result.numVirtual).to.be.undefined;
+      });
+  });
+
+  it('should not save virtual fields, but regenerate on retrieval', function() {
+    var Model = thinky.createModel(util.s8(), {
+      id: Number,
+      num: Number,
+      numVirtual: {
+        _type: 'virtual',
+        default: function() {
+          return this.num + 2;
+        }
+      }
+    });
+
+    var doc = new Model({
+      id: 1,
+      num: 1
+    });
+
+    expect(doc.numVirtual).to.equal(3);
+    return Model.ready()
+      .then(function() {
+        return doc.save();
+      })
+      .then(function(result) {
+        expect(result.numVirtual).to.equal(3);
+        return Model.get(1).execute();
+      })
+      .then(function(result) {
+        expect(result.numVirtual).to.be.undefined;
```

Comment:
This is insane to me. While I can see how this work, I believe most of the people don't know how this could `to.be.undefined` could work.
I don't see how this is a usual pattern in JS.